image-hd.c

```diff
@@ -26,6 +26,7 @@
 
 struct hdimage {
 	cfg_bool_t partition_table;
+	int32_t primary_partitions;
```

Comment:
just use 'unsigned int' here. Using `int32_t` only makes sense for values that must be 32 bit wide (e.g. the disk signature).